src/psbt.c

```diff
@@ -0,0 +1,2302 @@
+
+    // Global tx
+    type = WALLY_PSBT_GLOBAL_UNSIGNED_TX;
+    p += varbuff_to_bytes(&type, 1, p);
+    ret = wally_tx_get_length(psbt->tx, 0, &tx_len);
+    if (ret != WALLY_OK) {
+        return ret;
+    }
+    p += varint_to_bytes(tx_len, p);
+    ret = wally_tx_to_bytes(psbt->tx, 0, p, end - p, &tx_len);
+    if (ret != WALLY_OK) {
+        return ret;
+    }
+    p += tx_len;
+
+    // Unknowns
+    if (psbt->unknowns) {
+        for (i = 0; i < psbt->unknowns->num_items; ++i) {
+            struct unknowns_item *unknown = &psbt->unknowns->items[i];
+            p += varint_to_bytes(unknown->key_len, p);
+            memcpy(p, unknown->key, unknown->key_len);
+            p += unknown->key_len;
+            p += varint_to_bytes(unknown->value_len, p);
+            memcpy(p, unknown->value, unknown->value_len);
+            p += unknown->value_len;
+        }
+    }
+
+    // Separator
+    *p = WALLY_PSBT_SEPARATOR;
+    p++;
+
+    // Get lengths of each input and output
+    for (i = 0; i < psbt->num_inputs; ++i) {
+        struct wally_psbt_input *input = &psbt->inputs[i];
+        size_t input_len;
+        ret = psbt_input_to_bytes(input, p, end - p, &input_len);
+        if (ret != WALLY_OK) {
+            return ret;
+        }
+        p += input_len;
+    }
+    for (i = 0; i < psbt->num_outputs; ++i) {
+        struct wally_psbt_output *output = &psbt->outputs[i];
+        size_t output_len;
+        ret = psbt_output_to_bytes(output, p, &output_len);
+        if (ret != WALLY_OK) {
+            return ret;
+        }
+        p += output_len;
+    }
+    *bytes_written = p - bytes_out;
+    return WALLY_OK;
+}
+
+int wally_psbt_from_base64(
+    const char *string,
+    struct wally_psbt **output)
+{
+    char* decoded;
+    size_t safe_len;
+    ssize_t decoded_len;
+
+    // Allocate the decoded buffer
+    safe_len = base64_decoded_length(strlen(string));
+    if ((decoded = wally_malloc(safe_len)) == NULL) {
+        return WALLY_ENOMEM;
+    }
+
+    // Decode the base64 psbt
+    decoded_len = base64_decode(decoded, safe_len, string, strlen(string));
+    if (decoded_len <= 5) { // Make sure we also have enough bytes for the magic
+        return WALLY_EINVAL;
+    }
+
+    // Now decode the psbt
+    return wally_psbt_from_bytes((unsigned char*)decoded, decoded_len, output);
+}
+
+int wally_psbt_to_base64(
+    struct wally_psbt *psbt,
+    char **output)
+{
+    unsigned char* buff;
+    char *result = NULL;
+    size_t len, written, b64_safe_len;
+    ssize_t b64_len;
+    int ret = WALLY_OK;
+
+    if (!output) {
+        return WALLY_EINVAL;
+    }
+
+    if ((ret = psbt_get_length(psbt, &len)) != WALLY_OK) {
+        return ret;
+    }
+    if ((buff = wally_malloc(len)) == NULL) {
+        return WALLY_ENOMEM;
+    }
+
+    // Get psbt bytes
+    if ((ret = wally_psbt_to_bytes(psbt, buff, len, &written)) != WALLY_OK) {
+        goto done;
+    }
+
+    // Base64 encode
+    b64_safe_len = base64_encoded_le
```

Comment:
appears to assume psbts is never null.

Reply:
Added a check earlier to ensure this assumption is always true